packages/core/upload/admin/src/components/TableList/PreviewCell.js

```diff
@@ -1,69 +1,81 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
-import { prefixFileUrlWithBackendUrl, pxToRem } from '@strapi/helper-plugin';
-import { Avatar } from '@strapi/design-system/Avatar';
-import { Flex } from '@strapi/design-system/Flex';
-import { Typography } from '@strapi/design-system/Typography';
+import { getFileExtension, prefixFileUrlWithBackendUrl, pxToRem } from '@strapi/helper-plugin';
+import { Avatar, Initials } from '@strapi/design-system/Avatar';
+import { Box } from '@strapi/design-system/Box';
 import { Icon } from '@strapi/design-system/Icon';
 import Folder from '@strapi/icons/Folder';
 
-const GenericAssetWrapper = styled(Flex)`
-  span {
-    /* The smallest fontSize in the DS is not small enough in this case */
-    font-size: ${pxToRem(10)};
+import { AssetType } from '../../constants';
+import { createAssetUrl } from '../../utils';
+import { VideoPreview } from '../AssetCard/VideoPreview';
+
+const VideoPreviewWrapper = styled(Box)`
+  figure {
+    width: 26px;
+    height: 26px;
+  }
+
+  canvas,
+  video {
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+    border-radius: 50%;
   }
 `;
 
-export const PreviewCell = ({ alternativeText, fileExtension, mime, thumbnailURL, type, url }) => {
+export const PreviewCell = ({ type, element }) => {
   if (type === 'folder') {
     return (
-      <Flex
-        background="secondary100"
-        height={pxToRem(26)}
-        justifyContent="center"
-        width={pxToRem(26)}
-        borderRadius="50%"
-      >
-        <Icon color="secondary500" as={Folder} />
-      </Flex>
+      <Initials background="secondary100" textColor="secondary600">
+        <Icon color="secondary500" width={pxToRem(15)} height={pxToRem(15)} as={Folder} />
+      </Initials>
     );
   }
 
-  if (mime.includes('image')) {
-    const mediaURL = prefixFileUrlWithBackendUrl(thumbnailURL) ?? prefixFileUrlWithBackendUrl(url);
+  const { alternativeText, ext, formats, mime, name, url } = element;
+
+  if (mime.includes(AssetType.Image)) {
+    const mediaURL =
+      prefixFileUrlWithBackendUrl(formats?.thumbnail?.url) ?? prefixFileUrlWithBackendUrl(url);
 
     return <Avatar src={mediaURL} alt={alternativeText} preview />;
   }
 
+  if (mime.includes(AssetType.Video)) {
+    return (
+      <VideoPreviewWrapper>
+        <VideoPreview
+          url={createAssetUrl(element, true)}
+          mime={mime}
+          onLoadDuration={() => {}}
+          alt={alternativeText ?? name}
```

Comment:
I think we should not render `name` as fallback. We've moved away from that because it does more harm to a11y than good. Screenreaders will read the URL and therefore the filename as fallback if no alt attribute was set.

Oh and the `VideoPreview` implementation of that is wrong 🤦🏼 I think the only way to properly set an `alt` text for videos is to set a title attribute, rather than a figurecaption. An alternative could be to render the alt-text as child `<p>` of `<video>`. Both are out of scope for this PR but a good follow-up I'd say :)

Reply:
mmh, I think the difference here (for both `VideoPreview` used in grid and table) is that at the end it does not render a video tag but a canvas - I guess the idea was not to have a functional/playable video but an image to render in place of a video file:
<img width="369" alt="image" src="https://user-images.githubusercontent.com/71838159/212912772-1e3c3554-083a-4294-9a9d-edd289600e43.png">

doe `figure` and `figcaption` make sense in that usecase then?

Reply:
Oh sorry I didn't see that mess:

```js
const handleTimeUpdate = (e) => {
    if (e.target.currentTime > 0) {
      const video = e.target;
      const canvas = document.createElement('canvas');

      canvas.height = video.videoHeight;
      canvas.width = video.videoWidth;
      canvas.getContext('2d').drawImage(video, 0, 0, video.videoWidth, video.videoHeight);

      video.replaceWith(canvas);
      onLoadDuration(video.duration);
    }
  };
```

Wow - that is something. Wondering what the macOS screenreader reads in this case? 

The internet suggests to use `role=img` on a canvas element and `aria-label={alt}` - let's create a follow-up for this.

Reply:
👌
quick follow up, I actually get the right label with macOS screenreader 👀
<img width="960" alt="image" src="https://user-images.githubusercontent.com/71838159/212915958-44874252-493f-4091-ae74-b07101a88fa3.png">

until we find a better alternative I don't think I should remove the name fallback here as the figcaption wouldn't render anything and this information wouldn't be read anymore - wdyt?

Reply:
I agree.